### PR TITLE
Chromium updates with GN build

### PR DESCRIPTION
**Pull request by aszlig**
This should update Chromium to the latest versions along with the transition to GN and split up flash/widevine plugins.

However, the **flash plugin currently doesn't work** and I have no time digging through this properly, so if anyone got time to fix it, feel free to merge these with or without the fix of the flash plugin. The fix should be quite easy, but if it should get more involved I think it's better to have a broken flash plugin than a browser with tons of CVEs out there.

The work includes part of my initiel transition to the GN build system and a modified cc-wrapper from @bendlas to handle `@response` files during ninja build.

I've updated the Chromium versions to the latest ones, build and tests can be found here:

https://headcounter.org/hydra/eval/338754
(At the time of writing this, the builds aren't finished yet)

Note that the tests for beta and dev will probably fail, another issue which can wait for a while (until 55 becomes stable).

Cc: @grahamc,, @rycee, @FRidh, @fpletz, @NeQuissimus, @vcunat (participants in #19884 and #20078)

### CODE REVIEW

**Comment by sternenseemann** [comment]
Build failing because `pkg-config` is missing `gtk-3.0`:

```
configuring
ERROR at //build/config/linux/pkg_config.gni:85:17: Script returned non-zero exit code.
    pkgresult = exec_script(pkg_config_script, args, "value")
                ^----------
Current dir: /tmp/nix-build-chromium-56.0.2906.0.drv-0/chromium-56.0.2906.0/out/Release/
Command: python -- /tmp/nix-build-chromium-56.0.2906.0.drv-0/chromium-56.0.2906.0/build/config/linux/pkg-config.py gmodule-2.0 gtk+-3.0 gthread-2.0
Returned 1.
stderr:

Package gtk+-3.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `gtk+-3.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'gtk+-3.0' found
Could not run pkg-config.

See //build/config/linux/gtk3/BUILD.gn:14:1: whence it was called.
pkg_config("gtk3_internal_config") {
^-----------------------------------
See //chrome/browser/ui/libgtkui/BUILD.gn:143:12: which caused the file to be included.
           "//build/config/linux/gtk3",
           ^--------------------------

builder for ‘/nix/store/a10v51vzpw9wxi3pdhh781xjx7jwixdy-chromium-56.0.2906.0.drv’ failed with exit code 1
```

File: pkgs/applications/networking/browsers/chromium/common.nix [hunk 1]
```diff
@@ -195,22 +165,17 @@ let
...
-      echo "Precompiling .py files to prevent race conditions..." >&2
-      python -m compileall -q -f . > /dev/null 2>&1 || : # ignore errors
-
       # This is to ensure expansion of $out.
       libExecPath="${libExecPath}"
-      python build/linux/unbundle/replace_gyp_files.py ${gypFlags}
-      python build/gyp_chromium -f ninja --depth . ${gypFlags}
+      python build/linux/unbundle/replace_gn_files.py \
+        --system-libraries ${toString gnSystemLibraries}
+      gn gen --args=${escapeShellArg gnFlags} out/Release
```
Comment (sternenseemann) [inline pkgs/applications/networking/browsers/chromium/common.nix]: I think the mentioned build failure is triggered here, but I am not quite sure.

**Comment by bendlas** [comment]
@sternenseemann good to see testing of chromium-dev, but this PR is mostly about chromium-stable, since we need to fix those CVEs. Your build failure should be easily fixable, by adding gtk3 to buildInputs, though.

**Comment by sternenseemann** [comment]
@bendlas Oh, I see, totally missed that. The stable builds seem to be still in queue, sorry!

**Comment by grahamc** [comment]
This looks really good, but I'm afraid of breaking flash for our users, since I want to backport this to 16.09. This is a tough call for me. Having updates which break important features can easily turn users off from upgrading.

**Comment by grahamc** [comment]
@aszlig just a thought -- will it fail to build? or will it silently fail? Very different user experiences here.

**Comment by aszlig** [comment]
@grahamc: It won't fail to build and it just doesn't work. I think it's just a matter of passing the right parameters in the wrapper.

**Comment by bendlas** [comment]
Good news, after having installed this build to my system profile, flash actually works.
I don't know how or why, when I ran chromium from `./result` after a `nix-build`, flash didn't work, but now it does ...

**Comment by grahamc** [comment]
Building to give it a go.

**Comment by grahamc** [comment]
These changes also fix my weird yellow box issues (https://github.com/NixOS/nixpkgs/issues/1795,) so that is good.

I'm finding WideVine is now broken, not showing up in chrome://components/. Is this expected?

**Comment by grahamc** [comment]
Ported to 16.09 for testing here: https://github.com/NixOS/nixpkgs/pull/20120

**Comment by grahamc** [comment]
I'm not feeling too great about this, but not great about leaving it unpatched either. Merging. @aszlig can you check out my branch for 16.09 to see if it is correctly backported?

**Comment by aszlig** [comment]
@grahamc: I have time for the whole night to work on this now, so I could work on this along with fixing/testing WideVine/Flash and also incorporating the latest stdenv merge.

**Pull request closed without merging**